### ISSUE TITLE
why my loss is nan [I use the data set in the link that you provide」  

### ISSUE THREAD

**Issue by jiyongma** [divamgupta/image-segmentation-keras#43]
THEANO_FLAGS=device=gpu,floatX=float32  python  train.py  --save_weights_path=weights/ex1  --train_images="data/dataset2/images_prepped_train/"  --train_annotations="data/dataset2/annotations_prepped_train/"  --val_images="data/dataset2/images_prepped_test/"  --val_annotations="data/dataset2/annotations_prepped_test/"  --n_classes=2  --input_height=256 --input_width=320 --model_name="vgg_segnet" 

512/512 [==============================] - 49s - loss: nan - acc: 0.9032 - val_loss: nan - val_acc: 0.8787
Epoch 1/1
512/512 [==============================] - 49s - loss: nan - acc: 0.9005 - val_loss: nan - val_acc: 0.8800
Epoch 1/1
512/512 [==============================] - 49s - loss: nan - acc: 0.9017 - val_loss: nan - val_acc: 0.8788
Epoch 1/1
512/512 [==============================] - 49s - loss: nan - acc: 0.9033 - val_loss: nan - val_acc: 0.8804
Epoch 1/1
512/512 [==============================] - 50s - loss: nan - acc: 0.9006 - val_loss: nan - val_acc: 0.8791
Epoch 1/1
512/512 [==============================] - 49s - loss: nan - acc: 0.9056 - val_loss: nan - val_acc: 0.8805
Epoch 1/1

**Comment by basgro** [comment]
The test dataset has --n_classes=10 while you use --n_classes=2. I am not sure how keras computes the loss for annotated classes which are neither 0 or 1.
